# HG changeset patch
# User Marco Caliari <marco.caliari@univr.it>
# Date 1332708819 25200
#      Sun Mar 25 13:53:39 2012 -0700
# Node ID 319660795df10298396500e09085fd70111820e6
# Parent  351ca094580b47dcb513993643c15068a94823d8
Fix gnuplot error when color values are within eps of each other (bug #35761).

* plot/private/__go_draw_axes__.m: Use %.15e for setting cbrange in gnuplot.

diff --git a/scripts/plot/private/__go_draw_axes__.m b/scripts/plot/private/__go_draw_axes__.m
--- a/scripts/plot/private/__go_draw_axes__.m
+++ b/scripts/plot/private/__go_draw_axes__.m
@@ -1429,20 +1429,20 @@ function __go_draw_axes__ (h, plot_strea
     if (! any (isinf (clim)))
       if (truecolor || ! cdatadirect)
         if (rows(addedcmap) > 0)
           for i = 1:data_idx
             if (have_3d_patch(i))
               data{i}(end,:) = clim(2) * (data{i}(end, :) - 0.5) / cmap_sz;
              endif
           endfor
-          fprintf (plot_stream, "set cbrange [%g:%g];\n", clim(1), clim(2) *
+          fprintf (plot_stream, "set cbrange [%.15e:%.15e];\n", clim(1), clim(2) *
                    (cmap_sz + rows(addedcmap)) / cmap_sz);
         else
-          fprintf (plot_stream, "set cbrange [%g:%g];\n", clim);
+          fprintf (plot_stream, "set cbrange [%.15e:%.15e];\n", clim);
         endif
       else
         fprintf (plot_stream, "set cbrange [1:%d];\n", cmap_sz +
                  rows (addedcmap));
       endif
     endif
 
     if (strcmpi (axis_obj.box, "on"))
